### ISSUE TITLE
Couldn't Reproducing Numbers from Paper

### ISSUE THREAD

**Issue by Externalhappy** [zhengli97/PromptKD#7]
Hello,
I am trying to reproduce the accuracy in 11 datasets in your paper. After running PromptSrc to produce the teacher model, I renamed the model and moved it to the teacher_model folder. Then, I ran the script under promptkd base2new_train.sh. After finishing training, I tried to assess the accuracy on base and novel classes, and I found out that parse_test_rest gives the same number for both new and novel, which is the novel accuracy? Even the numbers are far away from the number delivered in the paper?

**Comment by Externalhappy** [comment]
I used the publicly available pre-trained models. For most datasets, the results align with those mentioned in the paper. However, for the 'dtd' dataset, there are discrepancies. Here are the results for three separate runs:.
                         run1	run2	         run3	Base	         run1	run2	        run3	        New	         HM
PromptKD	83.40	73.10	81.80	79.43	70.20	62.10	66.70	66.33	72.29.
It's worth noting that the results for seed 2 differ significantly and do not harmonize with the other outcomes. I've attempted multiple runs, but consistently obtained the same results.

**Comment by Externalhappy** [comment]
I used the publicly available pre-trained models. For most datasets, the results align with those mentioned in the paper. However, for the 'dtd' dataset, there are discrepancies. Here are the results after three separate runs:

	                Base	         New	HM
PromptKD	83.47	68.93	75.51
in Paper           85.84        71.37       77.94			
the results for old and  New differ significantly. I've tried multiple runs, but consistently obtained the same results

**Comment by zhengli97** [comment]
Maybe you can try using KD_WEIGHT=200.
In my experiment, I ran it four times, and the results are as follows:
Run 1: base:85.88, novel:71.25
Run 2: base: 86.69, novel: 71.37
Run 3: base: 85.76, novel: 71.86
Run 4: base: 85.88, novel: 71.01
The accuracy of the base class in the second experiment is too high, so I do not use this result. I take the average of the first, third, and fourth results.

Actually, it is hard to say why you cannot reproduce my results. I guess it may be caused by the computational environment. When I tried to reproduce the results of PromptSRC, my results were lower than reported. But when my classmate repeated it, he got normal results.
